qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1898,6 +1898,64 @@ def count_ops_longest_path(self):
                 op_dict[name] += 1
         return op_dict
 
+    def _get_qubit_input_output_node(self, qubit, in_or_out=False):
+        """Returns qubit and input or output node from an index."""
+        nodes = self.output_map if in_or_out else self.input_map
+        # Check if index is passed
+        if isinstance(qubit, int):
+            if qubit >= self.num_qubits():
+                raise DAGCircuitError(f"Qubit index {qubit} is out of range")
+            qubit = list(nodes.keys())[qubit]
+        if qubit not in self.qubits:
+            raise DAGCircuitError("Qubit was not found in circuit")
+        return (qubit, nodes.get(qubit, None))
+
+    def get_qubit_input_node(self, qubit):
+        """Returns qubit and input node from a qubit index."""
+        return self._get_qubit_input_output_node(qubit)
+
+    def get_qubit_output_node(self, qubit):
+        """Returns qubit and output node from a qubit index."""
+        return self._get_qubit_input_output_node(qubit, True)
```

Comment:
I don't think that these are actually needed for the implementation, if you see my comment below if you take in a `Qubit` object directly instead of a positional index you don't have to do an extra lookup here.

Reply:
I agree, these are not strictly needed. I just wanted to add them for ease of use. But they're not necessary for implementation. I will have those removed soon.